src/model.hpp

```diff
@@ -153,7 +185,11 @@ class model {
             PE("stepping", "communication");
 
             PE("exchange");
-            auto local_spikes = previous_spikes().gather();
+            auto local_spikes = previous_spikes().gather();            
+            std::sort(local_spikes.begin(), local_spikes.end(),
```

Comment:
Could add it to the communicator class. Also the connection sorting could be put together so we don't sort either if we don't need to.